CMakeLists.txt

```diff
@@ -43,7 +43,7 @@ option(JSON_Diagnostics                    "Use extended diagnostic messages." O
 option(JSON_ImplicitConversions            "Enable implicit conversions." ON)
 option(JSON_LegacyDiscardedValueComparison "Enable legacy discarded value comparison." OFF)           
 option(JSON_Install                        "Install CMake targets during install step." ${MAIN_PROJECT})
-option(JSON_MultipleHeaders                "Use non-amalgamated version of the library." OFF)
+option(JSON_MultipleHeaders                "Use non-amalgamated version of the library." ON)
```

Comment:
No reason other than that I think it makes more sense as the default. `-DJSON_MultipleHeaders=ON` was used inconsistently in `ci.cmake`. When there's an error the diagnostics are always more useful when they are attributed to specific headers instead of the amalgamated one.

To make sure there are no issues with the amalgamated header, I added a dedicated build step to `ci_test_amalgamation`.

Reply:
Since the default affects projects using the library via CMake, I think we should keep it using the amalgamated header for the time being.

I agree the multiple-header version is more helpful when it comes to diagnostics (and therefore should be used everywhere in the CI), but rather not surprise users like this.

Reply:
> Since the default affects projects using the library via CMake, I think we should keep it using the amalgamated header for the time being.

That's precisely why I changed the default. It only affects users of CMake and, most importantly, not in a functional way, except that it yields better diagnostics. What's the downside? Conan seems to already default to the more sensible `-DJSON_MultipleHeaders=ON` as we've learned recently.

> I agree the multiple-header version is more helpful when it comes to diagnostics (and therefore should be used everywhere in the CI), but rather not surprise users like this.

In what way would that surprise users?

Reply:
You're right. Let's do it - but not forget it in the ChangeLog.

Reply:
Ok. :-)